invenio/modules/oauthclient/contrib/orcid.py

```diff
@@ -122,12 +122,43 @@ def account_info(remote, resp):
 
 def account_setup(remote, token):
     """Perform additional setup after user have been logged in."""
+    from invenio.ext.cache import cache
+    from invenio.modules.accounts.models import UserEXT
     from invenio.ext.sqlalchemy import db
     from ..handlers import token_session_key
 
+    import json
+    import requests
+
+    orcid = session.get(token_session_key(remote.name) +
+                        "_account_info").get("external_id")
+
     extra_data = {
-        "orcid": session.get(token_session_key(remote.name) +
-                             "_account_info").get("external_id")
+        "orcid": orcid
         }
     token.remote_account.extra_data = extra_data
     db.session.commit()
+
+    # Fill user full name if not already set
+    user = UserEXT.query.filter_by(id=orcid).first().user
+    if user and not any([user.given_names, user.family_name]):
+        # Query ORCID to get the real name
+        request_url = 'http://orcid.org/{}/orcid-bio'.format(orcid)
```

Comment:
In the future it might be useful to use a wrapper such as https://github.com/scholrly/orcid-python instead of bare HTTP requests. I'm not saying the project is the right one, but we could contribute to it, or create our own ORCID wrapper.

At least for the GitHub integration on Zenodo it makes life a lot easier to have a wrapper.